### PR TITLE
merging new updates for command line single tests, and snapshots

### PR DESCRIPTION
**Pull request by SwingCoder911**
- enabling cli single test run
- snapshot
- upload to gcs

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -13,38 +12,24 @@ if [[ ! -e $f ]]; then
...
-node --abort-on-uncaught-exception --unhandled-rejections=strict $f $browser
+node --abort-on-uncaught-exception --unhandled-rejections=strict ./mozlenium.js -browser $browser
 res=$?
 etime=$(date +%s%3N)
 
 ((runtime=etime-stime))
 
-echo "Check finished in ${runtime:-0} ms with status code $res"
-echo TELEMETRY: node_time ${runtime:-0}
+echo "[STATUSMSG]: Check finished in ${runtime:-0} ms with status code $res"
```
Comment (mozafrank) [inline entrypoint.sh:32]: This format should be [TELEMETRY|status_msg]:

File: entrypoint.sh [hunk 2]
```diff
@@ -13,38 +12,24 @@ if [[ ! -e $f ]]; then
...
-if [[ ${WRAPPER:-1} -eq 1 ]]; then
-  # the wrapper edition!
-  echo 'async function runner($browser, $driver, $secure) {' > check.js
-  cat $f | grep -v "require('mozlenium')();" | sed 's|^$browser\.|return $browser.|' >> check.js
-  echo -e '}\n\nmodule.exports = runner;' >> check.js
-  f=wrapper.js
-fi
-
+export CHECKS_BASEDIR=/checks/
+export GCS_BUCKET=mozlenium-dev-snapshots
```
Comment (mozafrank) [inline entrypoint.sh:20]: This should be removed from here, and the node tooling should properly do nothing if this is not defined.

File: entrypoint.sh [hunk 3]
```diff
@@ -13,38 +12,24 @@ if [[ ! -e $f ]]; then
...
-
-if [[ ${WRAPPER:-1} -eq 1 ]]; then
-  # the wrapper edition!
-  echo 'async function runner($browser, $driver, $secure) {' > check.js
-  cat $f | grep -v "require('mozlenium')();" | sed 's|^$browser\.|return $browser.|' >> check.js
-  echo -e '}\n\nmodule.exports = runner;' >> check.js
-  f=wrapper.js
-fi
-
+export CHECKS_BASEDIR=/checks/
```
Comment (mozafrank) [inline entrypoint.sh:19]: lets remove this from here and have node default this to "./"

File: mozlenium-logger.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Class: MozleniumLogger
+ * This class's job is to provide an interface for logging well formatted logs
+ */
+class MozleniumLogger {
+  static ERROR_PREFIX = '[ERROR]: ';
+  static IMAGE_PREFIX = '[STATUSIMG]: ';
```
Comment (mozafrank) [inline mozlenium-logger.js:7]: lets change this to "[TELEMETRY|status_img]: "

File: mozlenium-logger.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  static TELEMETRY_PREFIX = 'TELEMETRY';
+  static LOG_PREFIX = '[MESSAGE]: ';
+  static STATUS_PREFIX = '[STATUSMSG]: ';
```
Comment (mozafrank) [inline mozlenium-logger.js:10]: "[TELEMETRY|status_msg]: "

File: mozlenium-logger.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Class: MozleniumLogger
+ * This class's job is to provide an interface for logging well formatted logs
+ */
+class MozleniumLogger {
+  static ERROR_PREFIX = '[ERROR]: ';
+  static IMAGE_PREFIX = '[STATUSIMG]: ';
+  static TELEMETRY_PREFIX = 'TELEMETRY';
```
Comment (mozafrank) [inline mozlenium-logger.js:8]: All telemetry messages should be of the format [TELEMETRY|<key>]: <val>
where val can be number or string w/whitespace,

File: entrypoint.sh (hunk 1 above)
Comment (SwingCoder911) [reply entrypoint.sh]: Addressed

File: entrypoint.sh (hunk 2 above)
Comment (SwingCoder911) [reply entrypoint.sh]: Addressed

File: mozlenium-logger.js (hunk 4 above)
Comment (SwingCoder911) [reply mozlenium-logger.js]: Addressed

File: mozlenium-logger.js (hunk 6 above)
Comment (SwingCoder911) [reply mozlenium-logger.js:7]: Addressed

**Pull request merged**